include/views/view.h

```diff
@@ -97,13 +97,17 @@ struct VectorView {
   template <bool use_as_ptr = false>
   SYCL_BLAS_INLINE typename std::enable_if<!use_as_ptr, value_t &>::type eval(
       index_t i) {
-    return (strd_ == 1) ? *(data_ + i) : *(data_ + i * strd_);
+    return (strd_ == 1)  ? *(data_ + i)
+           : (strd_ > 0) ? *(data_ + i * strd_)
+                         : *(data_ + (size_ * -strd_) + ((i + 1) * strd_));
```

Comment:
Maybe, we could add `size_ * -strd_` to the pointer at the construction of the of `VectorView` once or store this value in the `struct` as an `offset` variable rather than performing this multiplication every time we call the `eval` function.

Reply:
This will introduce two nested if statement which affect the performance.